hack/generator/pkg/astmodel/testdata/ConvertBetweenOptionalEnumAndOptionalBaseType.golden

```diff
@@ -0,0 +1,43 @@
+// Code generated by k8s-infra. DO NOT EDIT.
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package vCurrent
+
+import "github.com/Azure/k8s-infra/hack/generated/_apis/Verification/vNext"
+
+type Person struct {
+	sku *string `json:"sku"`
+}
+
+// ConvertFromVNext populates our Person from the provided source Person
+func (person Person) ConvertFromVNext(source *vNext.Person) error {
+
+	// sku
+	var sku string
+	if source.sku != nil {
+		sku = string(*source.sku)
+	} else {
+		sku = ""
+	}
+	person.sku = &sku
```

Comment:
If `source.sku` is nil then `person.sku` should be nil rather than `&""`, right?

Reply:
Yes - this is corrected in PR #433. To illustrate:

```
func (person Person) ConvertFromVNext(source *vNext.Person) error {

	// sku
	if source.sku != nil {
		sku := string(*source.sku)
		person.sku = &sku
	} else {
		person.sku = nil
	}

	// No error
	return nil
}
```